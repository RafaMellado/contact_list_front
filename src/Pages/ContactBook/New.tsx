import { FormEvent, useRef } from "react";
import { Button, Form } from "react-bootstrap";
import { useTranslation } from "react-i18next";
import { useNavigate } from "react-router-dom";
import ContactBookService from "../../Services/ContactBookService";

export function New() {
  const navigate = useNavigate();
  const { t } = useTranslation();

  const name = useRef<HTMLInputElement>(null);

  const TRANSLATIONS: string = "contactBookNew";

  const addContactBook = (event: FormEvent<HTMLFormElement>) => {
    event.preventDefault();

    ContactBookService.new({ name: String(name?.current?.value) });

    navigate("/home");
  };

  return (
    <>
      <Form onSubmit={addContactBook}>
        <Form.Group className="mb-3">
          <Form.Label>{t<string>(`${TRANSLATIONS}.name`)}</Form.Label>
          <Form.Control
            as="input"
            type="text"
            placeholder={t<string>(`${TRANSLATIONS}.name`)}
            maxLength={20}
            minLength={4}
            required
            ref={name}
          />
        </Form.Group>

        <Button variant="primary" type="submit">
          {t<string>(`${TRANSLATIONS}.submit`)}
        </Button>
      </Form>
    </>
  );
}
