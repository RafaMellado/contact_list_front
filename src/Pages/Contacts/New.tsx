import { useState } from "react";
import { useTranslation } from "react-i18next";
import { useNavigate, useParams } from "react-router-dom";
import ContactsService from "../../Services/ContactsService";
import {
  ContactRequestBody,
  ContactRequestError,
} from "../../Services/Interfaces/Contact";
import { ContactForm } from "./Components/Form";

export function ContactNew() {
  const { contactBookId } = useParams();
  const { t } = useTranslation();
  const navigate = useNavigate();
  const [errors, setErrors] = useState<ContactRequestError>();

  const TRANSLATIONS: string = "contactsNew";

  const backToList = () => {
    navigate(`/contact-book/show/${contactBookId}`);
  };

  const addContact = async (data: Partial<ContactRequestBody>) => {
    const newData = {
      ...data,
      contact_book_id: Number(contactBookId),
    };

    try {
      await ContactsService.create(newData as ContactRequestBody);

      backToList();
    } catch (errors) {
      setErrors(errors as ContactRequestError);
    }
  };

  return (
    <div className="d-flex flex-column align-items-center">
      <h2 className="mb-4">{t<string>(`${TRANSLATIONS}.title`)}</h2>

      <ContactForm
        translations={TRANSLATIONS}
        errors={errors}
        backToListFn={backToList}
        submitFn={addContact}
      />
    </div>
  );
}
