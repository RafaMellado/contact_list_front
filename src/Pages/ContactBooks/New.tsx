import { useTranslation } from "react-i18next";
import { useNavigate } from "react-router-dom";
import ContactBooksService from "../../Services/ContactBooksService";
import { ContactBookRequestBody } from "../../Services/Interfaces/ContactBook";
import { ContactBookForm } from "./Components/Form";

export function ContactBookNew() {
  const navigate = useNavigate();
  const { t } = useTranslation();
  const TRANSLATIONS: string = "contactBooksNew";

  const backToHome = () => {
    navigate("/home");
  };

  const addContactBook = async (data: ContactBookRequestBody) => {
    await ContactBooksService.create(data);

    backToHome();
  };

  return (
    <div className="d-flex flex-column align-items-center">
      <h2 className="mb-4">{t<string>(`${TRANSLATIONS}.title`)}</h2>

      <ContactBookForm
        translations={TRANSLATIONS}
        submitFn={addContactBook}
        backFn={backToHome}
      />
    </div>
  );
}
