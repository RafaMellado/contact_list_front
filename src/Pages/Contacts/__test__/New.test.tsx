import { ContactNew } from "../New";
import { act, render } from "@testing-library/react";
import { Router } from "react-router-dom";
import { createMemoryHistory } from "history";

jest.mock("react-i18next", () => ({
  useTranslation: () => ({ t: (key: string) => key }),
}));

describe(`<${ContactNew.name} />`, () => {
  const history = createMemoryHistory();

  const factoryComponent = () =>
    render(
      <Router location="/" navigator={history}>
        <ContactNew />
      </Router>
    );

  test("renders correctly the component", () => {
    const component = factoryComponent();

    act(() => {
      expect(component.asFragment()).toMatchSnapshot();
    });
  });
});
