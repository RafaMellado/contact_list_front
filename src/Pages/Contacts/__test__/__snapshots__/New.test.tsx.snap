// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ContactNew /> renders correctly the component 1`] = `
<DocumentFragment>
  <div
    class="d-flex flex-column align-items-center"
  >
    <h2
      class="mb-4"
    >
      contactsNew.title
    </h2>
    <div
      class="form card"
    >
      <div
        class="card-body"
      >
        <form
          class=""
        >
          <div
            class="mb-3"
          >
            <label
              class="form-label"
            >
              contactsNew.givenname
            </label>
            <input
              class="form-control"
              maxlength="24"
              minlength="2"
              placeholder="contactsNew.givenname"
              required=""
              type="text"
              value=""
            />
          </div>
          <div
            class="mb-3"
          >
            <label
              class="form-label"
            >
              contactsNew.surname
            </label>
            <input
              class="form-control"
              maxlength="24"
              minlength="2"
              placeholder="contactsNew.surname"
              required=""
              type="text"
              value=""
            />
          </div>
          <div
            class="mb-3"
          >
            <label
              class="form-label"
            >
              contactsNew.email
            </label>
            <input
              class="form-control"
              placeholder="contactsNew.email"
              required=""
              type="email"
              value=""
            />
          </div>
          <div
            class="mb-3"
          >
            <label
              class="form-label"
            >
              contactsNew.phone
            </label>
            <input
              class="form-control"
              maxlength="12"
              placeholder="contactsNew.phone"
              required=""
              type="tel"
              value=""
            />
          </div>
          <div
            class="d-flex justify-content-between"
          >
            <button
              class="w-100 btn btn-secondary"
              type="button"
            >
              contactsNew.backBtn
            </button>
            <button
              class="w-100 ms-3 btn btn-primary"
              type="submit"
            >
              contactsNew.submit
            </button>
          </div>
        </form>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
